api/overdrive.py

```diff
@@ -920,11 +920,14 @@ def update_licensepool_with_book_info(self, book, license_pool, is_new_pool):
         edition, is_new_edition = self._edition(license_pool)
 
         # If the pool does not already have a presentation edition,
-        # and if this edition is newly made, then associate pool and edition
-        # as presentation_edition
-        if ((not license_pool.presentation_edition) and is_new_edition):
+        # then associate pool and edition as presentation_edition.
+        if not license_pool.presentation_edition:
             edition_changed = license_pool.set_presentation_edition()
 
+        # Make sure the pool also has a work.
+        if not license_pool.work:
+            license_pool.calculate_work()
```

Comment:
Is there a better place to do this? In some cases this will get called when it was already getting called somewhere else, but it wasn't getting called at all for the problematic pools.